### PR TITLE
Account system (part)

### PR DESCRIPTION
**Pull request by jsun969**
Account system branch is too large. Merge it first and divide #9 todos to smaller issues and PRs.

CleanShot 2024-01-18 at 20 32 17@2x

### CODE REVIEW

**Comment by rayokamoto** [comment]
Yeah 51 commits is a bit crazy, ideally some of these commits should be squashed before it is merged. Or we can use the "squash and merge" strategy.

File: .env.local.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=YOUR_NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY
+CLERK_SECRET_KEY=YOUR_CLERK_SECRET_KEY
```
Comment (rayokamoto) [inline .env.local.example:2]: Add newline

**Review by rayokamoto — CHANGES_REQUESTED** [review]

File: src/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { authMiddleware } from '@clerk/nextjs';
+
+export default authMiddleware({
+    publicRoutes: [
+        '/',
+        '/not-found',
```
[the remark this answers is not in the document]
Comment (jsun969) [reply src/middleware.ts:6]: I'll create another issue for middleware. IMO it should be refactored.

File: src/components/UserButton.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+                {/* Popup menu */}
+                {isPopupOpen && (
+                    <div
+                        ref={popupRef}
+                        className="flex flex-col absolute w-52 md:w-44 gap-y-4 top-10 right-0 z-10 text-xl md:text-base bg-white p-4 border-4 border-black"
+                    >
+                        {/* Link to CS Club Drive */}
+                        <a
+                            href="https://csclub.us11.list-manage.com/track/click?u=17c2b13c547b901f8405e7bec&id=246194762c&e=acbfff1500"
```
[the remark this answers is not in the document]
Comment (phoenixpereira) [reply src/components/UserButton.tsx:39]: Will be addressed in a future PR.

**Review by phoenixpereira — APPROVED** [review]

**Review by rayokamoto — APPROVED** [review]

**Pull request merged**